### PR TITLE
Initial micro-ROS tutorials

### PR DESCRIPTION
**Pull request by pablogs9**

### CODE REVIEW

**Review by Acuadros95 — APPROVED** [review]

File: docs/rst/tutorials/micro/build_system_components/build_system_components.rst [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        ros2 run micro_ros_setup component arduino
+
+
+.. figure:: /rst/figures/tutorials/micro/platforms_logos/raspberry_pi_pico.png
+    :align: center
+
+Raspberry Pi Pico
+^^^^^^^^^^^^^^^^^
+
+`Raspberry Pi Pico <https://www.raspberrypi.com/products/raspberry-pi-pico/>`_ is the versatile embedded and super low cost solution of **Raspberry Pi for microcontroller environments**. It features a dual core Cortex-M0+ based silicon named RP2040, which include a wide variety of peripherals such as I2C, SPI, UART or GPIO, and even a hardware programmable module named PIO. Also a full featured API for hardware abstraction are provided within the SDK.
```
Comment (EduPonz) [inline docs/rst/tutorials/micro/build_system_components/build_system_components.rst:220]: I wouldn't expect "exagerating" adjectives such as `super`

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: docs/rst/tutorials/micro/build_system_components/build_system_components.rst [hunk 2]
```diff
@@ -0,0 +1,231 @@
+.. _tutorials_micro_build_system_components:
+
+Build System Components
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/tutorials/micro/build_system_components/build_system_components.rst:4]: 060d25cdfb32bbb2d3fde535d177158207af8e84

File: docs/rst/tutorials/micro/build_system_components/build_system_components.rst [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+.. figure:: /rst/figures/tutorials/micro/platforms_logos/zephyr_rtos.png
+    :align: center
+
+.. _micro_ros_zephyr_rtos:
+
+Zephyr RTOS
+^^^^^^^^^^^
+
+`Zephyr RTOS <https://docs.zephyrproject.org/>`_ is one of the most relevant RTOSes for embedded systems of the last years. **Supported by the Linux Foundation**, it provides a full featured and layered architectured RTOS ready for deploy production-ready applications. Thanks to the huge amount of ready-to-use drivers, the users will find easy to integrate connectivity solutions such as Bluetooth BLE, WiFi, USB or CAN.
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/tutorials/micro/build_system_components/build_system_components.rst]: 345efe1eeb01ea977ff6461af224178d0fff6152

[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/tutorials/micro/build_system_components/build_system_components.rst]: b0261f825d115eda82f22e4fda38c64405938508

File: docs/rst/tutorials/micro/build_system_components/build_system_components.rst [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+        source /opt/vulcanexus/humble/setup.bash
+        ros2 run micro_ros_setup component mbed_rtos
+
+.. figure:: /rst/figures/tutorials/micro/platforms_logos/nuttx_rtos.png
+    :align: center
+
+NuttX RTOS
+^^^^^^^^^^
+
+`NuttX <https://nuttx.apache.org/>`_ is one of the most complete and mature RTOS available. It is **supported by Apache Foundation** and provides a piece of software with emphasis on standard compliance and small footprint. It is a powerful and flexible RTOS that can be used for embedded applications and also is compliant with POSIX and ANSI standards.
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/tutorials/micro/build_system_components/build_system_components.rst]: 935c33f3d3959e06238c65a9b9d2dbbbbe668328

File: docs/rst/tutorials/micro/build_system_components/build_system_components.rst [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+
+micro-ROS app for NuttX provides an example application environment where the micro-ROS stack is integrated in the NuttX 10 build system and can be ran as a NuttX application with an UART transport.
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/tutorials/micro/build_system_components/build_system_components.rst]: a50307796e806ed7f57ca425a6dd93f706cf086a

File: docs/rst/tutorials/micro/build_system_components/build_system_components.rst (hunk 1 above)
Comment (pablogs9) [reply docs/rst/tutorials/micro/build_system_components/build_system_components.rst]: 0da9f37abd79499432ec12ba22303f687b90fe4d

File: docs/rst/tutorials/micro/introduction/introduction.rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+micro-ROS also targets **hard real-time** environments, giving the user control over:
+
+* Memory usage, achieving **zero dynamic memory usage** at runtime.
+* **Timing** of each operation to achieve **deterministic** behavior.
+* **Execution order** of the operations.
+
+micro-ROS is **transport agnostic**. With its default middleware (`eProsima Micro XRCE-DDS <https://micro-xrce-dds.docs.eprosima.com/en/latest/>`_) it is possible to run micro-ROS on top of almost any transport layer that supports a package or stream communication paradigm: TCP/IP or UDP/IP stacks, UART devices, CAN/FD buses, SPI, radio links, ...
+
+Architecture
+^^^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/tutorials/micro/introduction/introduction.rst:34]: What does this means?

File: docs/rst/tutorials/micro/introduction/introduction.rst [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+micro-ROS follows an architecture that is based on ROS 2, but also has significant differences that allows the embedded integration. In order to ease the use, it also provide tools for handling the integration in embedded platforms.
+
+micro-ROS **layered architecture** abstracts the OS/RTOS with a middleware layer. On top of the middleware, micro-ROS uses the same ROS 2 core layers (RMW, RCL and RCUTILS). With respect to user APIs, micro-ROS provides a **C99 API named RCLC** that provides the concepts available on ROS 2 C++ or Python APIs (RCLCPP or RCLPY).
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/tutorials/micro/introduction/introduction.rst]: 22e57010d1dd75e262e08e807cb99afd19836bc3

**Comment by pablogs9** [comment]
@EduPonz comments addressed. If this LGTY I'll fix CI

File: docs/rst/tutorials/micro/introduction/introduction.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/tutorials/micro/introduction/introduction.rst]: feb4649b17e9f0872453bfa9af05e522149f642f

File: docs/rst/tutorials/micro/introduction/introduction.rst [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+----------
+
+In the case of micro-ROS, the default middleware layer is `eProsima Micro XRCE-DDS <https://micro-xrce-dds.docs.eprosima.com/en/latest/>`_, a **lightweight**, **portable**,  with **minimal overhead** middleware based on the `OMG DDS-XRCE <https://www.omg.org/spec/DDS-XRCE/1.0/About-DDS-XRCE/>`_ standard.
+
+The micro-ROS default middleware is compound of two parts: the **Micro XRCE-DDS Client** that runs in the MCU side, and the **Micro XRCE-DDS Agent** that runs in the ROS 2 side and communicates the embedded side with the ROS 2 dataspace.
+The `eProsima Micro XRCE-DDS <https://micro-xrce-dds.docs.eprosima.com/en/latest/>`_ is the base package for the `micro-ROS Agent <https://github.com/micro-ROS/micro-ROS-Agent>`_, a package that extends the former with specific ROS 2 features such as graph management.
+
+.. note::
+
+    `micro-ROS Agent <https://github.com/micro-ROS/micro-ROS-Agent>`_ is distributed along with `Vulcanexus Micro <https://docs.vulcanexus.org/en/latest/>`_. Check the `installation tutorial <https://docs.vulcanexus.org/en/latest/rst/installation/linux_binary_installation.html>`_.
```
Comment (EduPonz) [inline docs/rst/tutorials/micro/introduction/introduction.rst:59]:
```suggestion
    `micro-ROS Agent <https://github.com/micro-ROS/micro-ROS-Agent>`_ is distributed along with `Vulcanexus Micro <https://docs.vulcanexus.org/en/latest/>`_.
    Check the `installation tutorial <https://docs.vulcanexus.org/en/latest/rst/installation/linux_binary_installation.html>`_.
```

File: docs/rst/tutorials/micro/build_system_components/build_system_components.rst [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+.. figure:: /rst/figures/tutorials/micro/platforms_logos/renesas_e2_studio.png
+    :align: center
+
+.. _micro_ros_renesas_e2_studio:
+
+Renesas e\ :sup:`2` Studio
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`Renesas RA Family <https://www.renesas.com/us/en/products/microcontrollers-microprocessors/ra-cortex-m-mcus>`_ is the **official supported hardware** of micro-ROS.
+This family of MCUs features a wide range of features based on Arm® Cortex®-M33, M23, and M4 processor cores.
```
Comment (EduPonz) [inline docs/rst/tutorials/micro/build_system_components/build_system_components.rst:22]: features features

File: docs/rst/tutorials/micro/build_system_components/build_system_components.rst [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+
+Renesas e\ :sup:`2` Studio provides a whole integrated development environment that allows for professional development, test and profiling of embedded applications.
+The micro-ROS component for Renesas provides a set of tools and instructions to integrate the build and configuration of micro-ROS inside of a Renesas e\ :sup:`2` Studio project.
```
Comment (EduPonz) [inline docs/rst/tutorials/micro/build_system_components/build_system_components.rst:25]:
```suggestion
The micro-ROS component for Renesas provides a set of tools and instructions to integrate the build and configuration of micro-ROS within a Renesas e\ :sup:`2` Studio project.
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: docs/rst/tutorials/micro/build_system_components/build_system_components.rst (hunk 9 above)
Comment (pablogs9) [inline docs/rst/tutorials/micro/build_system_components/build_system_components.rst:22]:
```suggestion
This family of MCUs provides a wide range of features based on Arm® Cortex®-M33, M23, and M4 processor cores.
```

**Review by EduPonz — APPROVED** [review]

**Pull request merged**